api/app/Http/Middleware/TransformInputDataMiddleware.php

```diff
@@ -27,8 +27,11 @@ public function handle($request, Closure $next)
             // Find any potential camelCase keys in the 'root' array, and convert
             // them to snake_case
             if (!ctype_lower($key)) {
-                $request->offsetSet(snake_case($key), $value);
-                $request->offsetUnset($key);
+                // Only convert if the key will change
+                if ($key != snake_case($key)) {
```

Comment:
Okay that makes sense, but wouldn't `ctype_lower('oauth_token')` return true so the block wouldn't be entered?

By the way the proxying from the frontend is just a simple proxypass, the actual frontend javascript client won't be aware of any of the exchange.